packages/cli/src/WorkflowHelpers.ts

```diff
@@ -524,3 +527,41 @@ export type NameRequest = Express.Request & {
 		name?: string;
 	};
 };
+
+/**
+ * Build a `where` clause for a `find()` or `findOne()` operation
+ * in the `shared_workflow` or `shared_credentials` tables.
+ */
+export function whereClause({
+	user,
+	entityType,
+	entityId = '',
+}: {
+	user: User;
+	entityType: 'workflow' | 'credentials';
+	entityId?: string;
+}): WhereClause {
+	const where: WhereClause = entityId ? { [entityType]: { id: entityId } } : {};
+
+	// TODO: Decide if owner access should be restricted
+	if (user.globalRole.name !== 'owner') {
+		where.user = { id: user.id };
+	}
+
+	return where;
+}
+
+/**
+ * Get the IDs of the workflows that have been shared with the user.
+ */
+export async function getSharedWorkflowIds(user: User): Promise<number[]> {
+	const sharedWorkflows = await Db.collections.SharedWorkflow!.find({
```

Comment:
We should return all workflow regardless of ownership if the current user is the instance owner.

Reply:
`getSharedWorkflowIds()` calls `whereClause()`, which adds the user ID filter only if the logged-in user is _not_ an owner.

https://github.com/n8n-io/n8n/blob/9ca13617d9aa574b47e0987054d3c16efdd68854/packages/cli/src/WorkflowHelpers.ts#L547-L549

I tested it with the owner and with a member (overriding as below), and it is returning all workflow IDs and the member's workflow IDs, respectively. Are you seeing different behavior? Or would you prefer for the owner's query not to go through `SharedWorkflow` at all, for performance?

```ts
req.user.id = 'someUserId';
req.user.globalRole.name = 'member';
req.user.globalRole.id = 3;
```

Reply:
If the latter, this might be best addressed when creating `UserManagementModel`.

Reply:
You are correct, thanks for pointing it out!